forum/users/api_view.py

```diff
@@ -514,3 +520,124 @@ def get_user_profile(self, access_token, userinfo_url):
+        """
+        logger.info("Entered ResetPasswordRequestView post method.")
+        email = request.data.get('email')
+        logger.info(f"Requested email: {email}")
+        
+        try:
+            user = User.objects.get(email=email)
+            logger.info(f"Queueing password reset email for user {user.email}")
+            send_reset_password_email.delay(user.user_id)
+            logger.info(f"Password reset email queued for {user.email}.")
+
+            return Response({"message": "Password reset email sent."}, status=status.HTTP_200_OK)
+        except User.DoesNotExist:
+            logger.warning(f"Password reset requested for non-existing email: {email}.")
+            return create_error_response("User with this email does not exist.", status.HTTP_404_NOT_FOUND)
+        except Exception as e:
+            logger.error(f"Failed to send password reset email to {email}: {e}")
+            return create_error_response("Failed to send password reset email.", status.HTTP_500_INTERNAL_SERVER_ERROR)
+
+
+class ResetPasswordConfirmView(APIView):
+    """
+    View to confirm password reset using a UID and token. 
+    This view accepts a POST request with a UID, token, and new password, 
+    verifies the token, and resets the user's password.
+    """
+    permission_classes = [AllowAny]
+    throttle_classes = [AnonRateThrottle]
+
+    def post(self, request, uidb64, token):
+        """
+        Handle POST request to reset the password for a given user.
+
+        Args:
+            request (Request): The request object containing the new password.
+            uidb64 (str): Base64 encoded user ID.
+            token (str): Password reset token.
+
+        Returns:
+            Response: A message indicating success or error response if token is invalid.
+        """
+        try:
+            uid = urlsafe_base64_decode(uidb64).decode()
+            user = User.objects.get(pk=uid)
+            if default_token_generator.check_token(user, token):
+                password = request.data.get('password')
+                user.set_password(password)
+                user.save()
+                logger.info(f"Password reset successfully for user ID: {user.user_id}.")
+                return Response({"message": "Password has been reset successfully."}, status=status.HTTP_200_OK)
+            else:
+                logger.warning(f"Invalid or expired token for user ID: {user.user_id}.")
+                return create_error_response("Invalid or expired token.", status.HTTP_400_BAD_REQUEST)
+        except (TypeError, ValueError, OverflowError, User.DoesNotExist) as e:
+            logger.error(f"Invalid token for password reset: {e}")
+            return create_error_response("Invalid token.", status.HTTP_400_BAD_REQUEST)
+        except Exception as e:
+            logger.error(f"Failed to reset password: {e}")
+            return create_error_response("Failed to reset password.", status.HTTP_500_INTERNAL_SERVER_ERROR)
```

Comment:
Further, validate the new password to enforce password policies (such as minimum length or complexity) before saving it.